zerver/lib/actions.py

```diff
@@ -2799,7 +2799,7 @@ def check_update_message(
     # sent the message, they are not the admin, and the time limit for editing
     # topics is passed, raise an error.
     if (
-        content is None
+        topic_name is not None
         and message.sender != user_profile
         and not user_profile.is_realm_admin
         and not is_no_topic_msg
```

Comment:
I think the error message string for this case might need to change somewhere in this PR; haven't checked if you've done so already.  Maybe "The time limit for editing this message's topic has passed".

Reply:
Fixed this in a separate commit.